libopenage/gamestate/terrain.h

```diff
@@ -9,42 +9,63 @@
 #include "util/vector.h"
 
 namespace openage {
-namespace renderer::terrain {
-class TerrainRenderEntity;
-}
+namespace renderer {
+class RenderFactory;
+} // namespace renderer
 
 namespace gamestate {
+class TerrainChunk;
 
 /**
  * Entity for managing the map terrain of a game.
  */
 class Terrain {
 public:
-	Terrain(const std::string &texture_path);
+	/**
+     * Create a new terrain.
+     */
+	Terrain();
 	~Terrain() = default;
 
 	/**
-	 * Set the current render entity of the terrain.
+     * Add a chunk to the terrain.
+     *
+     * @param chunk New chunk.
+     */
+	void add_chunk(const std::shared_ptr<TerrainChunk> chunk);
+
+	/**
+     * Get the chunks of the terrain.
+     *
+     * @return Terrain chunks.
+     */
+	const std::vector<std::shared_ptr<TerrainChunk>> &get_chunks() const;
+
+	/**
+	 * Attach a renderer which enables graphical display.
+     *
+     * TODO: We currently have to do attach this here too in addition to the terrain
+     *       factory because the renderer gets attached AFTER the terrain is
+     *       already created. In the future, the game should wait for the renderer
+     *       before creating the terrain.
```

Comment:
oooh I see what you mean now!

Reply:
fixed it. I think clang format misses these indentation because they are technically inside a comment. We could use `UseTab: Always` as a setting, although this could have unforseen consequences.